src/analysis.rs

```diff
@@ -12,7 +12,64 @@
 // If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 //
 
-use std::cmp;
+use std::{cmp, fmt};
+
+/// CPU cost of a Simplicity expression.
+///
+/// The cost is measured in milli weight units
+/// and can be converted into weight units using the appropriate method.
+///
+/// Roughly speaking, the operational semantics of a combinator
+/// on the Bit Machine determine its cost.
+///
+/// First, every combinator has a fixed overhead cost.
+/// Frame allocations, copy and write operations cost proportional
+/// to the number of allocated or written bits.
+/// Frame moves / drops or cursor moves are one-step operations
+/// that are covered by the overhead.
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct Cost(usize);
+
+impl Cost {
+    /// Overhead constant.
+    ///
+    /// Every combinator that is executed has this overhead added to its cost.
+    const OVERHEAD: Self = Cost(10);
+
+    /// Cost of combinators that are never executed.
+    ///
+    /// **This should only be used for `fail` nodes!**
+    const NEVER_EXECUTED: Self = Cost(0);
+
+    /// Convert the cost to weight units.
+    pub const fn to_weight(self) -> usize {
+        (self.0 + 999) / 1000
+    }
+
+    /// Return the cost of a type with the given bit width.
+    pub const fn from_type(bit_width: usize) -> Self {
+        Cost(bit_width)
+    }
+
+    /// Wrap the cost of a jet.
+    pub const fn from_jet(cost: usize) -> Self {
```

Comment:
In c45f8d3cc6d26c6435b3a3e06b4dc72d9a79e3b7:

You should call this `from_milliweight` and update the Simplicity PR accordingly.